pilopt/src/lib.rs

```diff
@@ -43,6 +48,43 @@ pub fn optimize<T: FieldElement>(mut pil_file: Analyzed<T>) -> Analyzed<T> {
     pil_file
 }
 
+fn hash_pil_state<T: FieldElement>(pil: &Analyzed<T>) -> u64 {
```

Comment:
Does it make sense to implement Hash for Analyzed? Would it be the correct implementation?

Reply:
It wasn't my intention, because there are several fields in `Analyzed<T>` that don't interest me (because they will not be modified during the optimizations) and hashing them would not add anything to the solution. 

Perhaps if another use case for the Analyzed hash appears, it might be worth using it here directly as well, but otherwise, I don't see the advantage.

Reply:
I'm a bit worried about us forgetting something here once we add it to Analyzed. The worry is almost as large when we implement Hash for Analyzed, but a bit smaller :)

What would be the fields that we don't need here?

Reply:
To be honest they are not a lot haha (`auto_added_symbols`, `source_order` & `solved_impls` right now).

Reply:
If the only advantage is performance I'd hash the whole thing since optimizer performance is not our bottleneck now.